superset/security/manager.py

```diff
@@ -2083,26 +2083,26 @@ def get_rls_filters(self, table: "BaseDatasource") -> list[SqlaQuery]:
         )
         return query.all()
 
-    def get_rls_ids(self, table: "BaseDatasource") -> list[int]:
+    def get_rls_filters_str(self, table: "BaseDatasource") -> list[str]:
         """
-        Retrieves the appropriate row level security filters IDs for the current user
-        and the passed table.
+        Retrieves the appropriate row level security filters string representation
+        (id concat'ed with clause) for the current user and the passed table.
 
         :param table: The table to check against
-        :returns: A list of IDs
+        :returns: A list of string representations of the user's RLS filters
         """
-        ids = [f.id for f in self.get_rls_filters(table)]
-        ids.sort()  # Combinations rather than permutations
-        return ids
+        filters = self.get_rls_filters(table)
+        filters.sort(key=lambda f: f.id)  # Combinations rather than permutations
+        str_reps = [f"{f.id}-{f.clause}" for f in filters]
```

Comment:
Probably a totally unnecessary perf optimization, but do we need the ids here?I think just using the clause and sorting them should be sufficient, and would avoid a cache miss if two filters are swapped (= id stays the same, but the clauses are interchanged).

Reply:
I included the ID here since I was thinking about an edge case where there could be two different RLS with the same clause but different type (regular/base) and we'd want to avoid a false positive cache hit.

However I just realized I was thinking about this wrong because the filter type only affects who the filter applies to, not how the clause is applied.  I'll remove the id and just sort by clause

Reply:
Ah, but now I've realized there's the group key which actually does affect how the clause is applied, so I think that does actually need to be considered for the cache key